test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -340,6 +340,27 @@ func Connect(mdb *mdbv1.MongoDBCommunity, opts *options.ClientOptions) error {
 	})
 }
 
+// Connect performs a connectivity check by initializing a mongo client
+// with the SRV connection string and inserting a document into the MongoDB
+// resource. Custom client options can be passed, for example to configure TLS.
+func ConnectSRV(mdb *mdbv1.MongoDBCommunity, opts *options.ClientOptions) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+	mongoClient, err := mongo.Connect(ctx, opts.ApplyURI(mdb.MongoSRVURI()))
+	if err != nil {
+		return err
+	}
+
+	return wait.Poll(time.Second*1, time.Second*30, func() (done bool, err error) {
+		collection := mongoClient.Database("testing").Collection("numbers")
+		_, err = collection.InsertOne(ctx, bson.M{"name": "pi", "value": 3.14159})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
```

Comment:
Just a passing thought: I think we have this exact piece of code replicated in a couple other places. Maybe worth a refactpr?

Reply:
Or maybe this is basically superseded by what @chatton said a few lines down